doc/api/n-api.md

```diff
@@ -3747,6 +3747,36 @@ napi_status napi_is_promise(napi_env env,
 - `[out] is_promise`: Flag indicating whether `promise` is a native promise
 object - that is, a promise object created by the underlying engine.
 
+### napi_get_promise_state
+<!-- YAML
+added: REPLACEME
+-->
+
+```C
+napi_status napi_get_promise_state(napi_env env,
+                                   napi_value promise,
+                                   napi_promise_state* state);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] promise`: The promise to examine.
+- `[out] state`: The current state of the promise.
+
```

Comment:
This is missing the "Returns" to match the standard doc format.

It would also be good to add an explanation of when/why you use the method